setup.py

```diff
@@ -305,10 +313,13 @@ def awscrt_ext():
         if not sys.platform.startswith('openbsd'):
             libraries += ['rt']
 
-        # hide the symbols from libcrypto.a
-        # this prevents weird crashes if an application also ends up using
-        # libcrypto.so from the system's OpenSSL installation.
-        extra_link_args += ['-Wl,--exclude-libs,libcrypto.a']
+        if using_system_libcrypto():
+            extra_link_args += ['-lcrypto']
+        else:
+            # hide the symbols from libcrypto.a
+            # this prevents weird crashes if an application also ends up using
+            # libcrypto.so from the system's OpenSSL installation.
+            extra_link_args += ['-Wl,--exclude-libs,libcrypto.a']
```

Comment:
can you verify that we are not leaking symbols when compiling against system libcrypto? we might want to keep symbol stripping behavior even for system crypto branch. (nm or objdump will let you dump symbols)